design/ceph/object/ceph-bucket-notification-crd.md

```diff
@@ -0,0 +1,62 @@
+# Ceph bucket notifications CRD
+
+## Overview
+
+In its new release, Ceph introduced the bucket notifications feature. It allows to send messages to various endpoints when a new event occurs on a bucket (ref: <https://docs.ceph.com/docs/master/radosgw/notifications/)>
+
+Setup of those notifications are normally done by sending HTTP requests to the RGW, either to create/delete topics pointing to specific endpoints, or create/delete bucket notifications based on those topics.
+
+This functionality eases this process by avoiding to use external tools or scripts. It is replaced by creation of CR definitions that  contain all the information necessary to create topics and/or notifications, which the rook operator processes.
+
+## Goals
+
+Creates a CRD for topics and a CRD for notifications, defining all the necessary and optional information for the various endpoints.
+
+Extends the rook operator to handle the CRs that would be submitted by users.
+
+## Implementation
+
+The CR for a topic configuration takes this form:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: Topic
+metadata:
+  name: # name of the topic
+  namespace: # namespace where topic belongs
+Spec:
+  Endpoint: #(mandatory) URI of an endpoint to send push notification to
+  Port: #(optional) port used to communicate, if empty uses the default ones
```

Comment:
port is part of the URI, don't think that we need a separate field

Reply:
so thats means `Port` value need to keep mandatory in the endpoint?

Reply:
just remove `Port`
the endpoint URI has an optional "port" section